agent/exec/controller.go

```diff
@@ -286,9 +286,7 @@ func Do(ctx context.Context, task *api.Task, ctlr Controller) (*api.TaskStatus,
 		status.PortStatus = portStatus
 	}()
 
-	// this branch bounds the largest state achievable in the agent as SHUTDOWN, which
-	// is exactly the correct behavior for the agent.
-	if task.DesiredState >= api.TaskStateShutdown {
+	if task.DesiredState == api.TaskStateShutdown {
```

Comment:
I wonder if this should be reverted or not, because making it `>=` is a bug fix that should have been in SwarmKit anyway. WDYT @anshulpundir?

Reply:
Agreed.

Reply:
@andrewhsu can you make sure we're not reverting this line then?